sphinx/screens/chats/chat-common/chat-common/src/main/java/chat/sphinx/chat_common/adapters/MessageListAdapter.kt

```diff
@@ -303,9 +303,9 @@ internal class MessageListAdapter<ARGS : NavArgs>(
 
                 includePaidMessageReceivedDetailsHolder.apply {
                     buttonPayAttachment.setOnClickListener {
-                        currentViewState?.let { viewState ->
-                            viewState.message?.let { message ->
-                                payAttachment(message, viewState)
+                        currentViewState?.message?.let { message ->
+                            onStopSupervisor.scope.launch(viewModel.mainImmediate) {
+                                viewModel.payAttachment(message)
```

Comment:
OnStopSupervisor scope is scoped to the lifecycle of the Fragment such that when it moves to `onStop()` all child coroutines are cancelled; it's use is primarily for observation of data.

As the method we are calling doesn't return any data (primarily because we use `ViewState`s to promote uni-directional data flow), coroutines are unnecessary here and their use can be moved to the ViewModel.